node/src/lib.rs

```diff
@@ -171,26 +257,99 @@ impl Node {
                 let heartbeat_url = {
                     let nodes = self.model.nodes().await;
                     let node_id = *nodes.iter().next().unwrap().0;
-
-                    let base = &registry.url.join("v1.0/").unwrap();
-                    base.join(&format!("health/nodes/{}", node_id)).unwrap()
+                    let registry = self.current_registry.lock().await.clone().unwrap();
+
+                    registry
+                        .url
+                        .join(&format!("{}/", self.api_version)) // Ensure it ends with a '/'
+                        .unwrap()
+                        .join(&format!("health/nodes/{}", node_id))
+                        .unwrap()
```

Comment:
who knew that .join() replaced the last segment of an URL instead of appending to it if it doesn't end in a '/'? I didn't, the author didn't, good to know... 😎